job-scheduler/src/main/java/org/hawkular/metrics/scheduler/impl/SchedulerImpl.java

```diff
@@ -361,7 +367,7 @@ public void start() {
                 queryScheduler)));
     }
 
-    private Observable<JobLock> acquireJobLock(JobDetails jobDetails) {
+    private Observable<JobLock> acquireJobLock(JobDetailsImpl jobDetails) {
```

Comment:
Even though this is an implementation for the scheduler, I would use the interface for job details.